### PR TITLE
Addition of raster functionality

### PR DESCRIPTION
**Pull request by kbonney**
## Summary
This PR adds the function `sample_raster` to the `wntr.gis` module. The function samples a raster file at the coordinate points of the provided geodataframe (e.g., junctions from WaterNetworkGIS). This PR only extends the existing API and does not modify it (ie, the inclusion of the PR merits a minor version change at most).

## Tests and documentation
One test is added to verify that the raster is correctly sampled. A docstring is included in the function definition. Currently no additional documentation is included, but there could be a subsection added to the **Geospatial capabilities** user manual page to discuss and exemplify usage of the new function.
 
## Acknowledgement
By contributing to this software project, I acknowledge that I have reviewed the [software quality assurance guidelines](https://wntr.readthedocs.io/en/stable/developers.html) and that my contributions are submitted under the [Revised BSD License](https://wntr.readthedocs.io/en/stable/license.html).

### CODE REVIEW

**Comment by kbonney** [comment]
@dbhart, can you review this?

**Review by dbhart — APPROVED** [review]
@kbonney Looks good to me!

**Comment by kbonney** [comment]
Discussion points:

- Do we want users to pass in the raster object or the filepath? I opted to have the user pass in the filepath, so that the function can take care of opening and closing the file. If we have users pass in the raster directly, they may forget to close the connection to the file. Other geospatial capabilities directly ingest the geospatial data, rather than filepaths, but geodataframes are different because they exist in memory whereas the raster object is a connection to the file in storage.
- Do we want to include any raster plotting functionality? For example, the function might take an existing `ax` with something like a water network plotted on it, extract the extent, fetch the corresponding raster data within the extent, and plot it using imshow. Something like this might be nice for the saltwater demo.
- User manual docs?

**Comment by dbhart** [comment]
Note to all - failed tests due to actions runner issue that still needs to be merged into main

**Comment by kbonney** [comment]
Ok, thanks @dbhart. Any thoughts on the first discussion point above?

**Comment by dbhart** [comment]
i think that we pass the filename. if the file stays open rather than loading the raster into memory then i agree, i think we want control over opening and close the file

**Comment by kbonney** [comment]
@kaklise,
Remade test to have verifiable sample values and to use Net1, also added the test as an example to the user docs.

File: documentation/gis.rst [hunk 1]
```diff
@@ -822,3 +829,100 @@ the census tracts (polygons) is different than the junction and pipe attributes.
...
+	
+    >>> wn = wntr.network.WaterNetworkModel('networks/Net1.inp') # doctest: +SKIP
+    >>> wn_gis = wntr.network.to_gis(wn, crs='EPSG:4326')
+
+Sample elevations at junctions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Elevation is an essential attribute for accurate simulation of pressure in a water network and is
+commonly provided in GeoTIFF (.tif) files. The following example shows how such files can be sampled 
+and assigned to the junctions and tanks of a network.
```
Comment (kaklise) [inline documentation/gis.rst:856]: Updating text to state that elevation data generally needs to be adjusted to account for buried pipes.

File: documentation/gis.rst [hunk 2]
```diff
@@ -822,3 +829,100 @@ the census tracts (polygons) is different than the junction and pipe attributes.
    :alt: Intersection of junctions and pipes with mean income demographic data in EPANET example Net1
 
    Net1 with mean income demographic data intersected with junctions and pipes.
+
+Sample raster at points geometries
+--------------------------------------
+
+The :class:`~wntr.gis.sample_raster` function can be used to sample a raster file at point geometries,
+such as the nodes of a water network. A common use case for this function is to assign elevation to the
```
Comment (kaklise) [inline documentation/gis.rst:837]: change "water network" to "water network model"

**Review by kaklise — CHANGES_REQUESTED** [review]
Looks good, just requesting minor updates to documentation

File: documentation/gis.rst (hunk 1 above)
Comment (kbonney) [reply documentation/gis.rst]: See latest commit

File: documentation/gis.rst (hunk 2 above)
Comment (kbonney) [reply documentation/gis.rst:837]: See latest commit

**Pull request merged**